### ISSUE TITLE
CPU spike

### ISSUE THREAD

**Issue by remmelt** [iverberk/nomad-ui#3]
Awesome UI!

On OSX Chrome, every request to the Nomad API is sent as fast as possible. This makes the CPU go crazy... I've also downloaded ~50MB of json in the time I've typed this ;)

Maybe a timeout?

**Comment by iverberk** [comment]
Hi Remmelt,

Thanks. I've seen this before and there actually already is a timeout for this but obviously it's not working :-) Do you happen to have a case for reproducing this? That would help a lot. Otherwise I might have to fallback to not automatically retry the requests and just let the user refresh the page.

**Comment by jippi** [comment]
i think its the `X-Nomad-Index` not being used correctly ... all my requests use ?index=1 or `? index=0` even though the header is in the API response and support blocking queries

**Comment by iverberk** [comment]
@jippi I'm not sure I follow. In what way is it being used incorrectly? When the UI starts it will query the nodes, allocs, evals and jobs with index=0. When this returns it will contain a new (higher) index which is used in the next request. In theory this should never cause an infinite loop unless the index remains the same and the call becomes non-blocking.

**Comment by remmelt** [comment]
Hi Ivo,

I can reproduce this like so:

Clone the nomad master git branch (5081b940e90fe80c3dfcfcc895598a7261edf85e)
`vagrant up`
(For me this didn't actually install nomad in the vm... so downloaded the nomad binary from the site)
Create `server.hcl`
Paste `http_api_response_headers { Access-Control-Allow-Origin = "*" }`

`./nomad agent -dev -config server.hcl`

I then had to create a tunnel to the VM
`ssh -L4646:localhost:4646 vagrant@127.0.0.1 -p 2222 -o Compression=yes -o DSAAuthentication=yes -o LogLevel=FATAL -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -i $PWD/nomad/.vagrant/machines/nomad-server01/virtualbox/private_key`

`docker run -e NOMAD_ADDR=localhost -p 8000:3000 iverberk/nomad-ui`

Check out the nomad agent logs.

**Comment by iverberk** [comment]
Okay, I think it start to make sense. I actually omitted a crucial piece of information in the readme on enabling CORS on the server. I've updated the readme with the information. Specifically, it is necessary to allow the x-nomad-index header to pass through. Hopefully that will resolve the spike. Although it should still adhere to the timeout, which it doesn't... Let me know how you get on with this.
